src/test/java/co/cask/hydrator/salesforce/etl/BaseSalesforceStreamingSourceETLTest.java

```diff
@@ -0,0 +1,180 @@
+
+    ProgramManager programManager =
+      deployETL(sourceConfig, sinkConfig, "SalesforceStreaming_" + name.getMethodName());
+    programManager.startAndWaitForRun(ProgramRunStatus.RUNNING, 30, TimeUnit.SECONDS);
+
+    // Wait for Spark to start up the job by checking if receiver thread exists
+    try {
+      Awaitility.await()
+        .atMost(SALESFORCE_RECEIVER_START_TIMEOUT_SECONDS, TimeUnit.SECONDS)
+        .pollInterval(SALESFORCE_RECEIVER_START_POLLING_INTERVAL_MS, TimeUnit.MILLISECONDS)
+        .until(this::hasReceiverThreadStarted);
+    } catch (ConditionTimeoutException e) {
+      throw new IllegalStateException(String.format("Salesforce receiver thread failed to start in %d seconds",
+                                                    SALESFORCE_RECEIVER_START_TIMEOUT_SECONDS), e);
+    }
+
+    SObject pushTopic = SalesforceStreamingSourceConfig.fetchPushTopicByName(partnerConnection, pushTopicName);
+    createdObjectsIds.add((String) pushTopic.getField("Id"));
+
+    return programManager;
+  }
+
+  protected String getOutputDatasetName() {
+    return "output-realtimesourcetest_" + name.getMethodName();
+  }
+
+  protected List<StructuredRecord> waitForRecords(ProgramManager programManager,
+                                                  int exceptedNumberOfRecords) throws Exception {
+    DataSetManager<Table> outputManager = getDataset(getOutputDatasetName());
+
+    try {
+      Awaitility.await()
+        .atMost(WAIT_FOR_RECORDS_TIMEOUT_SECONDS, TimeUnit.SECONDS)
+        .pollInterval(WAIT_FOR_RECORDS_POLLING_INTERVAL_MS, TimeUnit.MILLISECONDS)
+        .until(() -> MockSink.readOutput(outputManager).size() >= exceptedNumberOfRecords);
+    } catch (ConditionTimeoutException e) {
+      throw new IllegalStateException(
+        String.format("Timeout occured while waiting for %d records", exceptedNumberOfRecords), e);
+    }
+
+    programManager.stop();
+    programManager.waitForStopped(10, TimeUnit.SECONDS);
+    programManager.waitForRun(ProgramRunStatus.KILLED, 10, TimeUnit.SECONDS);
+
+    return MockSink.readOutput(outputManager);
+  }
+
+  private SparkManager deployETL(ETLPlugin sourcePlugin, ETLPlugin sinkPlugin, String appName) throws Exception {
+    ETLStage source = new ETLStage("source", sourcePlugin);
+    ETLStage sink = new ETLStage("sink", sinkPlugin);
+    DataStreamsConfig etlConfig = DataStreamsConfig.builder()
+      .addStage(source)
+      .addStage(sink)
+      .addConnection(source.getName(), sink.getName())
+      .setBatchInterval("1s")
+      .build();
+
+    AppRequest<DataStreamsConfig> appRequest = new AppRequest<>(APP_ARTIFACT, etlConfig);
+    ApplicationId appId = NamespaceId.DEFAULT.app(appName);
+    ApplicationManager applicationManager = deployApplication(appId, appRequest);
+
+    return applicationManager.getSparkManager(DataStreamsSparkLauncher.NAME);
+  }
+
+  private boolean hasReceiverThreadStarted() {
```

Comment:
We should not forget to think how to make same check in remoteInstance mode. Maybe wait for dummy message?